### PR TITLE
Ensure performance tests run for newly opened PRs

### PR DESCRIPTION
**Pull request by swissspidy**
This is a follow-up to [[56972]](https://core.trac.wordpress.org/changeset/56972) from core-58867

Turns out `github.event.before` is also "empty" containing all `00000` for the first commit in a new branch when opening a PR. That means when opening a PR performance tests don’t currently run until you add a second commit.

To avoid this scenario but still prevent an error when a new permanent branch like `6.4` is created, this changes the performance test workflow to simply skip the target comparison if there is no "before".

Trac ticket: https://core.trac.wordpress.org/ticket/59805

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by swissspidy** [comment]
Proof that the performance test ran for the very first commit in this PR, before I accidentally cancelled it by pushing another commit: https://github.com/WordPress/wordpress-develop/actions/runs/6771717029/job/18402712469?pr=5625

**Comment by swissspidy** [comment]
Looks like https://core.trac.wordpress.org/changeset/57085 fixed this

**Comment by desrosj** [comment]
@swissspidy Went to reopen this but looks like the branch was deleted. Do you want to undelete and reopen, or open a fresh PR?

File: .github/workflows/performance.yml [hunk 1]
```diff
@@ -180,6 +180,7 @@ jobs:
         run: node ./tests/performance/results.js
 
       - name: Check out target commit (target branch or previous commit)
+        if: ${{ ! contains( '00000000', env.TARGET_SHA ) }}
```
Comment (swissspidy) [inline .github/workflows/performance.yml]:
```suggestion
        if: ${{ ! contains( env.TARGET_SHA, '00000000' ) }}
```

File: .github/workflows/performance.yml [hunk 2]
```diff
@@ -189,17 +190,21 @@ jobs:
...
         run: npm run build
 
       - name: Run target performance tests (base/previous commit)
+        if: ${{ ! contains( '00000000', env.TARGET_SHA ) }}
         env:
           TEST_RESULTS_PREFIX: before
         run: npm run test:performance
 
       - name: Print target performance tests results
+        if: ${{ ! contains( '00000000', env.TARGET_SHA ) }}
```
Comment (swissspidy) [inline .github/workflows/performance.yml]:
```suggestion
        if: ${{ ! contains( env.TARGET_SHA, '00000000' ) }}
```

File: .github/workflows/performance.yml [hunk 3]
```diff
@@ -189,17 +190,21 @@ jobs:
...
       - name: Install npm dependencies
+        if: ${{ ! contains( '00000000', env.TARGET_SHA ) }}
         run: npm ci
 
       - name: Build WordPress
+        if: ${{ ! contains( '00000000', env.TARGET_SHA ) }}
         run: npm run build
 
       - name: Run target performance tests (base/previous commit)
+        if: ${{ ! contains( '00000000', env.TARGET_SHA ) }}
```
Comment (swissspidy) [inline .github/workflows/performance.yml]:
```suggestion
        if: ${{ ! contains( env.TARGET_SHA, '00000000' ) }}
```

File: .github/workflows/performance.yml [hunk 4]
```diff
@@ -189,17 +190,21 @@ jobs:
           git reset --hard $TARGET_SHA
 
       - name: Install npm dependencies
+        if: ${{ ! contains( '00000000', env.TARGET_SHA ) }}
         run: npm ci
 
       - name: Build WordPress
+        if: ${{ ! contains( '00000000', env.TARGET_SHA ) }}
```
Comment (swissspidy) [inline .github/workflows/performance.yml]:
```suggestion
        if: ${{ ! contains( env.TARGET_SHA, '00000000' ) }}
```

File: .github/workflows/performance.yml [hunk 5]
```diff
@@ -189,17 +190,21 @@ jobs:
           git reset --hard $TARGET_SHA
 
       - name: Install npm dependencies
+        if: ${{ ! contains( '00000000', env.TARGET_SHA ) }}
```
Comment (swissspidy) [inline .github/workflows/performance.yml]:
```suggestion
        if: ${{ ! contains( env.TARGET_SHA, '00000000' ) }}
```

**Comment by swissspidy** [comment]
No longer needed I think

**Pull request closed without merging**